templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml

```diff
@@ -86,6 +87,10 @@ contents:
 
           source /run/etcd/environment
 
+          set -a
+          source /run/etcd/etcd.conf
```

Comment:
I find the choice of `/run` here to be odd - are you explicitly choosing to have this config file wiped if the node reboots for another reason?  Shouldn't it be in `/etc/etcd/`¹?

In general a pattern I think we should take is to have the MCD generate/own "base config" files, and allow the admin to use "drop-ins" - this pattern is [supported by systemd](https://coreos.com/os/docs/latest/using-systemd-drop-in-units.html) for example.

It's pretty easy to implement in shell script which you already have here, just make it:

```
if -d /etc/etcd.d; then
  for x in /etc/etcd.d/*.conf; do
    source $x
  done
fi
```

¹ I put an etcd in your etc so you can config while you config

Reply:
> I find the choice of /run here to be odd - are you explicitly choosing to have this config file wiped if the node reboots for another reason?

Nice catch!! No this was an oversight, to be honest.

>> But by doing so because this config is monitored by MCO will result in eventually restarting the node.

>Is only kind of true today, we don't actually monitor any files actively. It would probably make sense to do so.

So if I lay down the conf with MCO and it changes on disk it will no cause restart or it will?

>  I put an etcd in your etc so you can config while you config

:) while I like your train of thought emulating systemd (did I say that?) I would really like a simple config setup that is familiar to 3.x devs. These configs should not change very often except during tuning or disaster? In this case I think a single config that a user does not worry about inheritance or  would be optimal. No matter what we say or do folks will adjust the MCO base config. Then send me a BZ because the base config was overridden by foobar.conf. We could iterate on that in the future?

Single config WYSIWYG is my vote but the floor is open.

Reply:
@abhinavdahiya I am curious about your feelings on this as well.

Reply:
> So if I lay down the conf with MCO and it changes on disk it will no cause restart or it will?

MCO doesn't watch the files on disk to see if they are being changed by non-config/non-MCO means.

Reply:
Ok that is not what I was understanding at all. If I adjust `/etc/kubernetes/manifests/etcd-member.yaml` my understanding is MCO will roll that out to the other nodes and eventually restart.

Reply:
IMO writing the default configuration for etcd using setup-etcd-env is not the best move forawrd, mostly as MC* already writes and manages config files for crio/kubelet both of which technically do not need the reboot to pickup the changes but still we go through the reboot cycle.
on the other hand we would need to restart the etcd pod to pick up new configuration. And we should use MC to roll those changes as that would give us a way to not roll invalid configuration to all the etcd nodes with quorum guard in place.
So I would move towards uniformity and hopefully when / if MCD gets do-not-reboot on certain we can piggyback.

on the single file vs systemd style, i'm leaning towards single file similar to kubelet.conf. dropins are not useful for this usecase.

i'm +1 on stetting up our etcd static pods to consume configurations from a file (I'm okay on any location)